pkg/diagnostics/cluster/roles.go

```diff
@@ -0,0 +1,93 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	kerrs "k8s.io/kubernetes/pkg/api/errors"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/rulevalidation"
+	osclient "github.com/openshift/origin/pkg/client"
+	policycmd "github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+// ClusterRoles is a Diagnostic to check that there is a working Docker registry.
+type ClusterRoles struct {
+	ClusterRolesClient osclient.ClusterRolesInterface
+	SARClient          osclient.SubjectAccessReviews
+}
+
+const (
+	ClusterRolesName = "ClusterRoles"
+)
+
+func (d *ClusterRoles) Name() string {
+	return ClusterRolesName
+}
+
+func (d *ClusterRoles) Description() string {
+	return "Check that the ClusterRoles are up-to-date"
+}
+
+func (d *ClusterRoles) CanRun() (bool, error) {
+	if d.ClusterRolesClient == nil {
+		return false, fmt.Errorf("must have client.ClusterRolesInterface")
+	}
+	if d.SARClient == nil {
+		return false, fmt.Errorf("must have client.SubjectAccessReviews")
+	}
+
+	return userCan(d.SARClient, authorizationapi.AuthorizationAttributes{
+		Verb:     "list",
+		Resource: "clusterroles",
+	})
+}
+
+func (d *ClusterRoles) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(ClusterRolesName)
+
+	reconcileOptions := &policycmd.ReconcileClusterRolesOptions{
+		Confirmed:  false,
+		Union:      false,
+		Out:        ioutil.Discard,
+		RoleClient: d.ClusterRolesClient.ClusterRoles(),
+	}
+
+	changedClusterRoles, err := reconcileOptions.ChangedClusterRoles()
+	if err != nil {
+		r.Error("ClusterRolesDiagnostic1000", err, fmt.Sprintf("Error inspecting ClusterRoles: %v", err))
```

Comment:
As previously noted, IDs don't have particular significance. In fact @liggitt wanted them to be devoid of meaning. Since any errors are going to show the diagnostic name anyway, this seems a little redundant, but I'm not going to defend the existing scheme either.